lib/spack/spack/solver/asp.py

```diff
@@ -3510,10 +3625,15 @@ def make_node(*, pkg: str) -> NodeArgument:
         """
         return NodeArgument(id="0", pkg=pkg)
 
-    def __init__(
-        self, specs: List[spack.spec.Spec], *, hash_lookup: Optional[ConcreteSpecsByHash] = None
-    ):
-        self._specs: Dict[NodeArgument, spack.spec.Spec] = {}
+    def __init__(self, specs, hash_lookup=None):
+        self._specs: Dict[NodeArgument, spack.spec.Spec]
+        self._specs = {}
+
+        # Matches parent nodes to splice node
+        self._splices: Dict[NodeArgument, List[Splice]]
+        self._splices = {}
+        self._splice_dag: Dict[NodeArgument, List[NodeArgument]]
+        self._splice_dag = {}
```

Comment:
Is there a reason these are all split into separate lines with the type annotation separated from the assignment?

Reply:
I think when I wrote that I was getting it confused with when you want to declare types for a loop variable, forgetting that you could inline them. Easy fix.